src/components/PostCommentForm.jsx

```diff
@@ -0,0 +1,46 @@
+import { addCommentByArticleId } from "../../api";
+import { useState } from "react";
+import { useContext } from "react";
+import { UserContext } from "../contexts/User";
+
+function PostCommentForm({ id, setComments }) {
+  const [newComment, setNewComment] = useState("");
+  const { user } = useContext(UserContext);
+
+  function handleSubmit(event) {
+    addCommentByArticleId(id, newComment, user)
+      .then((response) => {
+        alert(`You posted your comment!`);
+        setComments((currComments) => {
+          return [response.data.comment, ...currComments];
+        });
+      })
+      .catch((error) => {
+        alert(`Oopsie, there's been an error. Try again please.`);
+        console.log(error);
+      });
+
+    setNewComment("");
+    event.preventDefault();
+  }
+
+  return (
+    <form
+      className="form-container"
+      name="form-container"
+      onSubmit={handleSubmit}
+    >
+      <label htmlFor="form-container">Add a comment</label>
+      <textarea
+        value={newComment}
+        placeholder="be nice!"
+        onChange={(event) => {
+          setNewComment(event.target.value);
+        }}
+      ></textarea>
+      <button className="submit-button">Post</button>
```

Comment:
Currently I can click the submit button multiple times - can we disable it after the first click to prevent errors?